paywall/src/components/interface/Header.js

```diff
@@ -0,0 +1,201 @@
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import React, { useState, useCallback } from 'react'
+import styled from 'styled-components'
+import Link from 'next/link'
+import { WordMarkLogo } from './Logo'
+import Buttons from './buttons/layout'
+import { ButtonLink } from './buttons/Button'
+import Media from '../../theme/media'
+
+// add new navigation buttons here, layout will reflow appropriately
+const navigationButtons = [
+  Buttons.About,
+  Buttons.Jobs,
+  Buttons.Github,
+  Buttons.Telegram,
+]
+
+export const mapStateToProps = ({
+  router: {
+    location: { pathname },
+  },
+}) => ({ pathname })
+
+export function Header({ title }) {
+  const [menu, setMenu] = useState(false)
+  const toggleMenu = useCallback(() => setMenu(showing => !showing), [setMenu])
+  return (
+    <TopHeader>
+      <Title forContent>
+        <Link href="/">
+          <TitleLink>
+            <WordMarkLogo viewBox="0 0 1200 256" height="28px" name="Unlock" />
+          </TitleLink>
+        </Link>
+        <TitleText>{title}</TitleText>
+      </Title>
+      <DesktopButtons>
+        {navigationButtons.map(NavButton => (
+          <NavButton key={NavButton} />
+        ))}
+      </DesktopButtons>
+      <MobileToggle visibilityToggle={!!menu} onClick={toggleMenu}>
+        <Buttons.Bars size="48px" />
+        <Buttons.ChevronUp size="48px" />
+      </MobileToggle>
+      <MobilePopover visibilityToggle={!!menu}>
+        {menu
+          ? navigationButtons.map(NavButton => (
+              <NavButton key={NavButton} size="48px" onClick={toggleMenu} />
+            ))
+          : ''}
+      </MobilePopover>
+    </TopHeader>
+  )
+}
+
+Header.propTypes = {
+  title: PropTypes.string,
+}
+
+Header.defaultProps = {
+  title: 'Unlock',
+}
+
+export default connect(mapStateToProps)(Header)
```

Comment:
It looks like the paywall header doesn't need access to the router state, so could you disconnect this component and remove `mapStateToProps` and the associated test?

Reply:
done, thanks for the catch. PTAL @cnasc